hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/string/HazelcastPositionFunction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators.string;
+
+import com.hazelcast.sql.impl.calcite.validate.HazelcastCallBinding;
+import com.hazelcast.sql.impl.calcite.validate.operand.OperandCheckerProgram;
+import com.hazelcast.sql.impl.calcite.validate.operand.TypedOperandChecker;
+import com.hazelcast.sql.impl.calcite.validate.operators.ReplaceUnknownOperandTypeInference;
+import com.hazelcast.sql.impl.calcite.validate.operators.common.HazelcastFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+public class HazelcastPositionFunction extends HazelcastFunction {
+    public static final HazelcastPositionFunction INSTANCE = new HazelcastPositionFunction();
+
+    public HazelcastPositionFunction() {
+        super(
+                "POSITION",
+                SqlKind.OTHER_FUNCTION,
+                ReturnTypes.INTEGER_NULLABLE,
+                new ReplaceUnknownOperandTypeInference(SqlTypeName.VARCHAR),
+                SqlFunctionCategory.STRING
+        );
+    }
+
+    @Override
+    public String getSignatureTemplate(int operandCount) {
```

Comment:
I wonder why it's implemented in some functions while not in others. Is it actually used? Under what circumstances?

Reply:
I have put a breakpoint but it didn't get triggered in test cases.
After looking the code, `calcite.SqlCallBinding.newValidationSignatureError()` seems the only place indirectly reaches this call.
We have `HazelcastCallBinding` which overrides it, and Calcite code calls the overridden method.

Reply:
So, if I understand correctly it's never used. Maybe we should simply remove it (and all other overrides)?

Reply:
OK, I will remove `getSignatureTemplate`. 
I have added `unparse` override and what I have seen from the code, it is used in several places.